src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -113,6 +119,19 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
 
             return new BoundIsPatternExpression(
                 node, expression, pattern, decisionDag, whenTrueLabel: whenTrueLabel, whenFalseLabel: whenFalseLabel, boolType, hasErrors);
```

Comment:
> whenTrueLabel: whenTrueLabel, whenFalseLabel: whenFalseLabel [](start = 55, length = 61)

Are labels now "flipped around" for a negated pattern? This is very subtle and can be a source of confusion down the road. Consider capturing ```negated``` in the node and documenting its effect, or, perhaps, "flipping" the labels here explicitly.

Reply:
I considered adding a `bool IsNegated` property but that state is easily available from the `DecisionDag`. And switching the labels seem as fragile as leaving them as is.

---
In reply to: [530521673](https://github.com/dotnet/roslyn/pull/49369#discussion_r530521673) [](ancestors = 530521673)

Reply:
> I considered adding a bool IsNegated property but that state is easily available from the DecisionDag. 

I am not sure I am following. It doesn't look like the state is inferred from a DecisionDag in the lowering and in the nullable walker. In any case, I think we need to have that clearly documented. Consider adding a comment with a note above this construction and also adding comments in BoundNodes.xml for the properties involved.  

---
In reply to: [530599832](https://github.com/dotnet/roslyn/pull/49369#discussion_r530599832) [](ancestors = 530599832,530521673)